src/Infrastructure.VS.UnitTests/ActiveDocumentLocatorTests.cs

```diff
@@ -34,25 +36,57 @@ public class ActiveDocumentLocatorTests
     {
         private Mock<IVsMonitorSelection> monitorSelectionMock;
         private Mock<ITextDocumentProvider> textDocumentProviderMock;
+        private Mock<IThreadHandling> threadHandlingMock;
         private ITextDocument textDocument;
 
         private ActiveDocumentLocator testSubject;
 
         [TestInitialize]
         public void TestInitialize()
         {
-            ThreadHelper.SetCurrentThreadAsUIThread();
-
             monitorSelectionMock = new Mock<IVsMonitorSelection>();
             textDocumentProviderMock = new Mock<ITextDocumentProvider>();
             textDocument = Mock.Of<ITextDocument>();
+            
+            threadHandlingMock = new Mock<IThreadHandling>();
+            threadHandlingMock.Setup(x => x.RunOnUIThread(It.IsAny<Action>())).Callback<Action>(op => op());
```

Comment:
FYI, we have the class NoOpThreadHandling that essentially does exactly this.
You don't need to change it now but just FYI